src/hooks/useTxDetails.ts

```diff
@@ -0,0 +1,36 @@
+import { sameAddress } from '@/utils/addresses'
+import type { TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import { getTransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import { useRouter } from 'next/router'
+import useAsync from './useAsync'
+import useChainId from './useChainId'
+import useSafeInfo from './useSafeInfo'
+
+/**
+ * @returns TransactionDetails for transaction id in url
+ */
+export const useTxDetails = (): [TransactionDetails | undefined, Error | undefined, boolean] => {
+  const chainId = useChainId()
+  const router = useRouter()
+  const { id } = router.query
+  const transactionId = Array.isArray(id) ? id[0] : id
+  const { safe, safeAddress } = useSafeInfo()
```

Comment:
I'd suggest we make this hook more reusable and pass these as arguments. It's very reliant on the URL otherwise.

There are a few other places we could then replace current logic as well (out of scope for the time being though).

Reply:
I will change it so that we can pass an optional `transactionID` and if this isn't passed we fallback to the id from the router. 

Of course I will also document it as a comment.